js/messaging.js

```diff
@@ -0,0 +1,511 @@
+            window.location.replace("/index.html");
+        }
+    }).catch(function (err) {
+        // This code runs if there were any errors
+        console.log(err);
+    });
+}
+
+// -----------HTML HELPERS-----------
+function htmlEncode(value) {
+    // Create a in-memory element, set its inner text (which is automatically encoded)
+    // Then grab the encoded contents back out. The element never exists on the DOM.
+    return $("<textarea/>").text(value).html();
+}
+
+function htmlDecode(value) {
+    return $("<textarea/>").html(value).text();
+}
+// ---------------------------------
+
+// Try to fetch name:id directory
+function loadDirectory() {
+    $.ajax({
+        url: DIRECTORY_URL,
+        crossDomain: true,
+
+        dataType: "json",
+        cache: false,
+        type: "GET",
+
+        success: (data) => {
+            // If we were able to retrieve it, update the saved directory
+            localforage.setItem("directory", data);
+            directory = data;
+            // Populate autocomplete
+            populateAutocomplete();
+        },
+
+        error: () => {
+            // Otherwise, try to retrieve stored directory
+            localforage.getItem("directory").then((stored_directory) => {
+                if (stored_directory === null) {
+                    // If unable, set directory to null (so other functions know that we don't have it)
+					UIAlert( D("nameDirectoryNotSet"), "loadDirectory(): stored_directory === null" );
+                    directory = null;
+                    // Disable send button
+                    document.getElementById("msg-send").disabled = true;
+                } else {
+                    directory = stored_directory;
+                    // Populate autocomplete
+                    populateAutocomplete();
+                }
+            });
+        }
+    });
+}
+
+function populateAutocomplete() {
+    let elems = document.querySelectorAll(".autocomplete-fullname");
+    // če se uporablja globalna var directory, ki je shranjena kot objekt (vedno shranjen kot reference), bo pri let x=y x le pointer na object y
+    // in se bo spremenil z spremembo "originala". spodnja stvar itak ni preveč efficent, loop čez vseh 7000 ljudi bi lahko delal težave...
+    // kakšen Object.keys bi bila boljša varianta ampak raje napišem tale komentar... idk, to se mi je zdelo uporabno ampak sedaj obžalujem
+    // samo guglal sem "copying an object js"
+    let autocomplete_entries = Object.assign({}, directory);
+    for (let variableKey in autocomplete_entries) {
+        autocomplete_entries[variableKey] = null;
+    }
+
+    M.Autocomplete.init(elems, {
+        data: autocomplete_entries,
+        onAutocomplete: validateName,
+        minLength: 0
+    });
+
+    if(window.location.hash.length > 1) {
+    	$("#full-name").val(decodeURIComponent(window.location.hash.substring(1)));
+    } else {
+    	$("#full-name").val(getUrlParameter("m"));
+    }
```

Comment:
Je to backwards compatibility?
Bi se dalo uporabiti zgolj eno stvar (hash/param), ne obojega?